mqttclients/dotnet/MQTTnet.Client.Extensions/MqttConnectionSettings.cs

```diff
@@ -145,6 +145,26 @@ private static MqttConnectionSettings ParseConnectionString(string connectionStr
         return cs;
     }
 
+    private static int CheckForValidIntegerInput(string envVarName, string envVarValue)
+    {
+        if (int.TryParse(envVarValue, out var result))
+        {
+            return result;
+        }
+
+        throw new ArgumentException($"An environment variable's type was specified incorrectly: {envVarName}={envVarValue}");

```

Comment:
Can you include that it should be an int (and bool in the one below) in the error message, since we know what it should be?